packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -50,6 +63,76 @@ const TextStyle _kLargeTitleTextStyle = TextStyle(
   color: CupertinoColors.black,
 );
 
+// There's a single tag for all instances of navigation bars because they can
+// all transition between each other (per Navigator) via Hero transitions.
+const _HeroTag _heroTag = _HeroTag();
+
+class _HeroTag {
+  const _HeroTag();
+  // This class exists because the Gallery smoke test wants all widgets from
+  // debugDumpApp() to have a toString();
+  @override
+  String toString() => "Cupertino navigation bars' Hero tag";
+}
+
+TextStyle _navBarItemStyle(Color color) {
+  return new TextStyle(
+    fontFamily: '.SF UI Text',
+    fontSize: 17.0,
+    letterSpacing: -0.24,
+    color: color,
+  );
+}
+
+/// Returns `child` wrapped with background and a bottom border if background color
+/// is opaque. Otherwise, also blur with [BackdropFilter].
+Widget _wrapWithBackground({
+  Border border,
+  Color backgroundColor,
+  Widget child,
+  bool annotate = true,
+}) {
+  Widget result = child;
+  if (annotate) {
+    final bool darkBackground = backgroundColor.computeLuminance() < 0.179;
+    final SystemUiOverlayStyle overlayStyle = darkBackground
+        ? SystemUiOverlayStyle.light
+        : SystemUiOverlayStyle.dark;
+    result = new AnnotatedRegion<SystemUiOverlayStyle>(
+      value: overlayStyle,
+      sized: true,
+      child: result,
+    );
+  }
+  final DecoratedBox childWithBackground = new DecoratedBox(
+    decoration: new BoxDecoration(
+      border: border,
+      color: backgroundColor,
+    ),
+    child: result,
+  );
+
+  if (backgroundColor.alpha == 0xFF)
+    return childWithBackground;
+
+  return new ClipRect(
+    child: new BackdropFilter(
+      filter: new ImageFilter.blur(sigmaX: 10.0, sigmaY: 10.0),
+      child: childWithBackground,
+    ),
+  );
+}
+
+// Whether the current route supports nav bar hero transitions from or to.
+bool _isTransitionable(BuildContext context) {
+  final ModalRoute<dynamic> route = ModalRoute.of(context);
+
```

Comment:
there's only two lines in this function, it's not clear to me that we gain much from a blank line between them :-P
(up to you)